### PR TITLE
[INF] Update `mkdocstrings` version to 0.19.0

### PR DESCRIPTION
**Pull request by Zeroto521**
# PR Description

Please describe the changes proposed in the pull request:

Part of #1133.

When I removed this line (https://github.com/pyjanitor-devs/pyjanitor/commit/cfeaad775e796bb7d232a4b52eb3c1131a628f0b). the doc-preview ci still failed.
Due to mkdocstring which stays in a lower version.
So it is time to update it.

```diff
       - name: Build docs
         run: |
           conda activate pyjanitor-dev
-          python -m ipykernel install --user --name pyjanitor-dev
           pip install -e .
           mkdocs build
```

**This PR is related to https://github.com/pyjanitor-devs/pyjanitor/pull/1133#discussion_r939503090.**

# PR Checklist

Please ensure that you have done the following:

1. [x] PR in from a fork off your branch. Do not PR from `<your_username>`:`dev`, but rather from `<your_username>`:`<feature-branch_name>`.

2. [x] If you're not on the contributors list, add yourself to `AUTHORS.md`.

3. [x] Add a line to `CHANGELOG.md` under the latest version header (i.e. the one that is "on deck") describing the contribution.
    - Do use some discretion here; if there are multiple PRs that are related, keep them in a single line.

# Automatic checks

There will be automatic checks run on the PR. These include:

- Building a preview of the docs on Netlify
- Automatically linting the code
- Making sure the code is documented
- Making sure that all tests are passed
- Making sure that code coverage doesn't go down.

# Relevant Reviewers

Please tag maintainers to review.

- @ericmjl

### CODE REVIEW

File: environment-dev.yml [hunk 1]
```diff
@@ -54,3 +53,4 @@ dependencies:
   - xorg-libxrender
   - pip:
       - mknotebooks
+      - mkdocstrings[python]
```
Comment (ericmjl) [inline environment-dev.yml]: If possible, could we move `mkdocstrings` to the conda-installed section? I am still unsure if `pip` still respects `conda`-installed dependencies.

File: mkdocs.yml [hunk 2]
```diff
@@ -46,7 +46,7 @@ plugins:
             show_source: true
           # https://github.com/mkdocstrings/mkdocstrings/issues/278#issuecomment-831411193
           selection:
-            docstring_style: restructured-text
+            docstring_style: sphinx
```
Comment (ericmjl) [inline mkdocs.yml:49]: :tada:

File: CHANGELOG.md [hunk 3]
```diff
@@ -14,6 +14,7 @@
 -   [ENH] Add `names_expand` and `index_expand` parameters to `pivot_wider` for exposing missing categoricals. Issue #1108 @samukweku
 -   [ENH] Add fix  for slicing error when selecting columns in `pivot_wider`. Issue #1134 @samukweku
 -   [ENH] `dropna` parameter added to `pivot_longer`. Issue #1132 @samukweku
+-   [INF] Update `mkdocstrings` version and to fit its new coming features. PR #1138 @Zeroto521
```
Comment (ericmjl) [inline CHANGELOG.md:17]: Oooh, this is exciting! What new features are coming?

Comment (Zeroto521) [reply CHANGELOG.md:17]: Sine 0.19 mkdocstrings have to set a handler. This is a breaking changing.

But good for us, we can link to sphinx doc easily. https://mkdocstrings.github.io/usage/#cross-references-to-other-projects-inventories

File: environment-dev.yml (hunk 1 above)
Comment (Zeroto521) [reply environment-dev.yml]: I tried but failed https://github.com/pyjanitor-devs/pyjanitor/commit/8cf8609e37e960352d4cdfcaca9b8b20ccca19e8

Conda seems can't work well with extra installing like 'package[extra]'

Maybe we could install directly mksdocstrings and its extra package (mkdocstrings-python) via conda.

https://github.com/mkdocstrings/mkdocstrings/blob/252b9c6c2dd7ea9818db55ae85cb5d346732bedb/pyproject.toml#L44

Comment (Zeroto521) [reply environment-dev.yml]: fixed via installing `mkdocstrings-python`

**Comment by Zeroto521** [comment]
`softmax` tests failing is fixed via #1139

**Review by ericmjl — APPROVED** [review]
Approving! I am also going to take the liberty of merging, because the change is not in core code and is very small.

**Pull request merged**